File: app/src/main/java/com/botijasoftware/utils/Circle.java
package com.botijasoftware.utils;

public class Circle {
	
	public Circle(int px, int py, int r) {

			X = px;
			Y = py;
			radius = r;
	}
	
	public Circle(Vector2 position, int r) {

			X = (int) position.X;
			Y = (int) position.Y;
			radius = r;
	}
	
	public void setRadius(int r) {
		radius = r;
	}
	
	public void setPosition(int x, int y) {
		X = x;
		Y = y;
	}
	
	public void setPosition(Vector2 v) {
		X = (int) v.X;
		Y = (int) v.Y;
	}
	
	public void move(int incx, int incy) {
		X += incx;
		Y += incy;
	}
	
	public void move(Vector2 v) {
		X += (int)v.X;
		Y += (int)v.Y;
	}
	
	public boolean contains(int x, int y) {
		int dx = X - x;
		int dy = Y - y;
		return ( dx <= radius && dx >= -radius && dy <= radius && dy >= -radius);
	}
	
	public boolean contains(Vector2 v) {
		return contains((int)v.X, (int)v.Y);
	}
	
	public void setCenter(int x, int y) {
		X = x;
		Y = y;
	}
	
	public void setCenter(Vector2 v) {
		X = (int)v.X;
		Y = (int)v.Y;
	}
	
	public Vector2 getCenter() {
		return new Vector2( X, Y );
	}
	
	
	public int X = 0; //center
	public int Y = 0;  
	public int radius;

}

File: app/src/main/java/com/botijasoftware/utils/Circle.kt
package com.botijasoftware.utils

class Circle {

    var X = 0 //center
    var Y = 0
    var radius: Int = 0

    constructor(px: Int, py: Int, r: Int) {

        X = px
        Y = py
        radius = r
    }

    constructor(position: Vector2, r: Int) {

        X = position.X.toInt()
        Y = position.Y.toInt()
        radius = r
    }

   /* fun setRadius(r: Int) {
        radius = r
    }*/

    fun setPosition(x: Int, y: Int) {
        X = x
        Y = y
    }

    fun setPosition(v: Vector2) {
        X = v.X.toInt()
        Y = v.Y.toInt()
    }

    fun move(incx: Int, incy: Int) {
        X += incx
        Y += incy
    }

    fun move(v: Vector2) {
        X += v.X.toInt()
        Y += v.Y.toInt()
    }

    fun contains(x: Int, y: Int): Boolean {
        val dx = X - x
        val dy = Y - y
        return dx <= radius && dx >= -radius && dy <= radius && dy >= -radius
    }

    operator fun contains(v: Vector2): Boolean {
        return contains(v.X.toInt(), v.Y.toInt())
    }

    fun setCenter(x: Int, y: Int) {
        X = x
        Y = y
    }

    var center: Vector2
        get() = Vector2(X.toFloat(), Y.toFloat())
        set(v) {
            X = v.X.toInt()
            Y = v.Y.toInt()
        }


}
